aws-android-sdk-s3/src/main/java/com/amazonaws/services/s3/AmazonS3Client.java

```diff
@@ -2003,6 +2003,7 @@ public PutObjectResult putObject(PutObjectRequest putObjectRequest)
         result.setETag(returnedMetadata.getETag());
         result.setMetadata(returnedMetadata);
         result.setRequesterCharged(returnedMetadata.isRequesterCharged());
+        result.setContentMd5(contentMd5);
```

Comment:
can you add unit and/or integration test for contentMd5 to be null, non-null and contentMd5 passed through ObjectMetadata by the developer?